plugins/ltac/g_tactic.mlg

```diff
@@ -450,6 +450,11 @@ GRAMMAR EXTEND Gram
   ;
   as_or_and_ipat:
     [ [ "as"; ipat = or_and_intropattern_loc -> { Some ipat }
+      | "as"; ipat = equality_intropattern ->
+        { match ipat with
+          | IntroRewrite _ -> user_err Pp.(str "Disjunctive/conjunctive pattern expected")
+          | IntroInjection _ -> user_err Pp.(str "Found an injection pattern while a disjunctive/conjunctive pattern was expected; use tactic injection instead.")
```

Comment:
Could this produce the right syntax for injection, or would that be too hard?

Reply:
Currently, the error is produced even in multiple forms of `destruct`, as in `destruct n as [|n], m as [=]`, and actually also in `induction n as [=]`, etc. It is not clear to me what would be the replacement to suggest in general.

If only the `destruct c as [= ...]` form is supposed to suggest a replacement, that would require to put the error at another place, and it would somehow start to be quite invasive, compare to the current situation which already errors with an `or_and_intropattern_loc`-expected syntax error.

Reply:
I think it would be useful to at least give an example -- I knew about `injection` before, but this error message would have helped me as I did not know about `injection X as [=H1 H2]`.

Reply:
It sounds like giving a perfect suggestion is not worth it, but I like Ralf's proposal; but let's make sure the suggestion is not misleading.

Reply:
So, to make a concrete proposal what about:

> use "injection ... as <pattern>" instead.

Reply:
Done (w/o the `< ... >` though, which are not a usage in Coq messages).